### DISCUSSION TITLE
Typing a passed atom as prop

### DISCUSSION THREAD

**Discussion by mutheusalmeida** [pmndrs/jotai#1413] [Q&A]
Hi, guys. I'm passing an atom as prop to my component, how can I type the setter function? By the way, is totally fine to pass an atom as prop? Thanks in advance

**Comment by dai-shi** [comment]
It is totally fine to pass an atom as a prop.

You probably want `WritableAtom`?

**Reply by mutheusalmeida** [reply]
Yeah, I'm using it. But I give it an any because I don't know how to use it

**Reply by dai-shi** [reply]
Can you create an example to show your goal with https://www.typescriptlang.org/play/ ?

You can start from this: https://tsplay.dev/Nle4Rm
